src/XamlX.IL.Cecil/CecilEvent.cs

```diff
@@ -1,33 +1,39 @@
-using System.Collections.Generic;
 using Mono.Cecil;
-using XamlX.TypeSystem;
 
 namespace XamlX.TypeSystem
 {
     partial class CecilTypeSystem
     {
         class CecilEvent : IXamlEventInfo
         {
-            private readonly TypeReference _declaringType;
-            public CecilTypeSystem TypeSystem { get; }
+            private readonly CecilTypeResolver _typeResolver;
             public EventDefinition Event { get; }
 
-            public CecilEvent(CecilTypeSystem typeSystem, EventDefinition ev, TypeReference declaringType)
+            public CecilEvent(CecilTypeResolver typeResolver, EventDefinition ev)
             {
-                _declaringType = declaringType;
-                TypeSystem = typeSystem;
+                _typeResolver = typeResolver;
                 Event = ev;
             }
-            public bool Equals(IXamlEventInfo other) => other is CecilEvent cp && cp.Event == Event;
+
             public string Name => Event.Name;
 
             private IXamlMethod _getter;
 
             public IXamlMethod Add => Event.AddMethod == null
                 ? null
-                : _getter ?? (_getter = TypeSystem.Resolve(Event.AddMethod, _declaringType));
+                : _getter ??= new CecilMethod(_typeResolver, Event.AddMethod);
```

Comment:
So no way to cache those now, right?

Reply:
Also, how does it resolve generic arguments from declaring generic type instance now? IIRC EventDefinition is pointing to TypeDefinition metadata that lacks that information.

Reply:
> So no way to cache those now, right?

I don't think methods were cached previously? But yes, caching is hard.